include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -128,27 +129,137 @@ namespace LinearAlgebra
         resize_val(const types::global_dof_index new_alloc_size,
                    types::global_dof_index &     allocated_size,
                    ::dealii::MemorySpace::
-                     MemorySpaceData<Number, ::dealii::MemorySpace::Host> &data)
+                     MemorySpaceData<Number, ::dealii::MemorySpace::Host> &data,
+                   const MPI_Comm &comm_shared)
         {
-          if (new_alloc_size > allocated_size)
+          if (comm_shared == MPI_COMM_SELF)
             {
-              Assert(((allocated_size > 0 && data.values != nullptr) ||
-                      data.values == nullptr),
-                     ExcInternalError());
+              if (new_alloc_size > allocated_size)
+                {
+                  Assert(((allocated_size > 0 && data.values != nullptr) ||
+                          data.values == nullptr),
+                         ExcInternalError());
 
-              Number *new_val;
-              Utilities::System::posix_memalign(
-                reinterpret_cast<void **>(&new_val),
-                64,
-                sizeof(Number) * new_alloc_size);
-              data.values = {new_val, [](Number *data) { std::free(data); }};
+                  Number *new_val;
+                  Utilities::System::posix_memalign(
+                    reinterpret_cast<void **>(&new_val),
+                    64,
+                    sizeof(Number) * new_alloc_size);
+                  data.values = {new_val,
+                                 [](Number *data) { std::free(data); }};
 
-              allocated_size = new_alloc_size;
+                  allocated_size = new_alloc_size;
+                }
+              else if (new_alloc_size == 0)
+                {
+                  data.values.reset();
+                  allocated_size = 0;
+                }
+              data.values_sm = {
+                ArrayView<const Number>(data.values.get(), new_alloc_size)};
             }
-          else if (new_alloc_size == 0)
+          else
             {
-              data.values.reset();
-              allocated_size = 0;
+#ifdef DEAL_II_WITH_MPI
+#  if DEAL_II_MPI_VERSION_GTE(3, 0)
+              allocated_size = new_alloc_size;
+
+              const unsigned int size_sm =
+                Utilities::MPI::n_mpi_processes(comm_shared);
+              const unsigned int rank_sm =
+                Utilities::MPI::this_mpi_process(comm_shared);
+
+              MPI_Win mpi_window;
+              Number *data_this;
+
+
+              std::vector<Number *> others(size_sm);
+
+              MPI_Info info;
+              MPI_Info_create(&info);
+
+              const bool contiguous_allocation_enabled = false; // TODO
```

Comment:
what about this TODO?

Reply:
I have removed the TODO. In the future there might be cases where we might want to allocate the array in one chunk (e.g. in the context of `FE_Q`).